implementation/src/main/java/io/smallrye/openapi/runtime/io/OpenApiSerializer.java

```diff
@@ -918,19 +919,23 @@ private void writeHeader(ObjectNode parent, Header model, String name) {
             return;
         }
         ObjectNode node = parent.putObject(name);
-        JsonUtil.stringProperty(node, OpenApiConstants.PROP_$REF, model.getRef());
-        JsonUtil.stringProperty(node, OpenApiConstants.PROP_DESCRIPTION, model.getDescription());
-        JsonUtil.booleanProperty(node, OpenApiConstants.PROP_REQUIRED, model.getRequired());
-        JsonUtil.booleanProperty(node, OpenApiConstants.PROP_DEPRECATED, model.getDeprecated());
-        JsonUtil.booleanProperty(node, OpenApiConstants.PROP_ALLOW_EMPTY_VALUE, model.getAllowEmptyValue());
-        JsonUtil.enumProperty(node, OpenApiConstants.PROP_STYLE, model.getStyle());
-        JsonUtil.booleanProperty(node, OpenApiConstants.PROP_EXPLODE, model.getExplode());
-        writeSchema(node, model.getSchema(), OpenApiConstants.PROP_SCHEMA);
-        writeObject(node, OpenApiConstants.PROP_EXAMPLE, model.getExample());
-        writeExamples(node, model.getExamples());
-        writeContent(node, model.getContent());
-        writeExtensions(node, model);
 
+        if (StringUtils.isNotEmpty(model.getRef())) {
```

Comment:
So something like model.getRef()!=null && !model.getRef().isEmpty() ?

Reply:
Fair point, I've pushed changes to use a utility method rather than the Commons dependency.